SwissTransferFeatures/SettingsView/EditSettings/EditSettingView.swift

```diff
@@ -0,0 +1,70 @@
+/*
+ Infomaniak SwissTransfer - iOS App
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import InfomaniakDI
+import STCore
+import STResources
+import SwiftUI
+import SwissTransferCore
+import SwissTransferCoreUI
+
+struct EditSettingView<T: SettingSelectable>: View {
+    @EnvironmentObject private var mainViewState: MainViewState
+
+    let title: String
+    let section: String
+    let items: [T]
+    let selected: T
+    let onSelection: (T) async -> Void
```

Comment:
Why do we need a onSelection function?
You can just do `await item.setSelected()`